server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java

```diff
@@ -303,15 +307,14 @@ public boolean authorizeRequest(RequestDetails requestDetails) {
         proxyResponse.addHeader(header.getName(), header.getValue());
       }
       // This should be called after adding headers.
-      // TODO handle non-text responses, e.g., gzip.
       // TODO verify DEFAULT_CONTENT_TYPE/CHARSET are compatible with `entity.getContentType()`.
       Writer writer =
           proxyResponse.getResponseWriter(
               response.getStatusLine().getStatusCode(),
               response.getStatusLine().toString(),
               DEFAULT_CONTENT_TYPE,
               Constants.CHARSET_NAME_UTF8,
-              false);
+              sendGzippedResponse(servletDetails));
       Reader reader;
       if (content != null) {
```

Comment:
Just making sure: The `content` is created by `outcome.postProcess(response)` above and response comes from the `fhirClinet`; does `fhirClient` automatically handles `gzip`? IOW, the response from `fhirClient` is already unzipped?

Reply:
Yes, the Apache HTTP Client we use for `fhirClient` supports compression/decompression out of the box.